### PR TITLE
Horizontal scale

### PR DESCRIPTION
**Pull request by NataliAharoniPayu**

### CODE REVIEW

File: controllers/operator_entrypoints_module.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+func ClusterRebalance(c echo.Context) error {
+	if reconciler == nil || cluster == nil {
+		return c.String(http.StatusOK, "Could not perform cluster rebalance action")
+	}
+	reconciler.saveClusterStateView(cluster)
+	v, ok := reconciler.NewRedisClusterView(cluster)
+	if !ok {
+		return c.String(http.StatusOK, "Could not retrieve redis cluster view to")
```
Comment (NataliAharoniPayu) [inline controllers/operator_entrypoints_module.go:35]: typo, remove 'to' from message

File: controllers/operator_entrypoints_module.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	if !found {
+		return c.String(http.StatusOK, "Could not find healthy server to serve the fix request")
+	}
+	healthyServerIp := v.Nodes[healthyServerName].Ip
+	mutex := &sync.Mutex{}
+	mutex.Lock()
+	reconciler.RedisClusterStateView.ClusterState = view.ClusterFix
+	_, _, err := reconciler.RedisCLI.ClusterFix(healthyServerIp)
+	if err != nil {
+		reconciler.Log.Error(err, "Could not perform cluster fix")
```
Comment (NataliAharoniPayu) [inline controllers/operator_entrypoints_module.go:76]: logistical issue - after each fix it is recommended to run 'rebalance', we can reflect it in log or set the cluster state to 'rebalance' or do both by informing the user to pay attention that rebalance takes action in next reconcile loop, if it didn't  - suggest to run it manually with /rebalance

**Review by shyimo — CHANGES_REQUESTED** [review]
I think you did an amazing job so far. wall done @NataliAharoni99.

i try to tackle the operator in various way and all worked well. the code also looks really good in general and i don't have a lot of comments (which is usually not the case :) )

File: helm/values.yaml [hunk 3]
```diff
@@ -60,8 +60,8 @@ redisCluster:
   enabled: true
   name: dev-rdc
   namespace: default
-  leaderCount: 3
-  leaderFollowersCount: 1
+  leaderCount: 5
```
Comment (voltbit) [inline helm/values.yaml:63]: Lets keep this setup small by default.

File: helm/values.yaml [hunk 4]
```diff
@@ -60,8 +60,8 @@ redisCluster:
...
+  leaderFollowersCount: 2
```
Comment (voltbit) [inline helm/values.yaml:64]: Lets keep this setup small by default.

File: server/router.go [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
 
 func Register(e *echo.Echo) {
 	e.GET("/state", clusterState)
 	e.GET("/info", clusterInfo)
+	e.POST("/reset", controllers.DoResetCluster)
+	e.POST("/rebalance", controllers.ClusterRebalance)
+	e.POST("/fix", controllers.ClusterFix)
+	e.POST("/forgetLostNodes", controllers.ForgetLostNodes)
+	e.POST("/forceReconcile", controllers.ForceReconcile)
+	e.POST("/upgrade", controllers.UpgradeCluster)
```
Comment (voltbit) [inline server/router.go:16]: We need documentation for each endpoint.

File: main.go [hunk 6]
```diff
@@ -91,15 +92,18 @@ func startManager() {
...
 
 	if err = (&controllers.RedisClusterReconciler{
-		Client:   mgr.GetClient(),
-		Log:      rdcLogger,
-		Scheme:   mgr.GetScheme(),
-		RedisCLI: getRedisCLI(&rdcLogger),
-		Config:   &operatorConfig.Config,
-		State:    controllers.NotExists,
+		Client:                mgr.GetClient(),
+		Cache:                 mgr.GetCache(),
```
[the remark this answers is not in the document]
Comment (NataliAharoniPayu) [reply main.go:100]: It was used for a practice i considered worth to have - 'block' the k8s api request for pod list until cache is synced (wait for k8s cache sync is a method cache object has), eventually a different practice was chosen: to have the request on time and wait for terminating pods if appeared during pod status analyze. 
as i see it, the second choice is covering better the range of cases that requires waiting prior to operator action.
im removing those to lines.

File: server/router.go [hunk 7]
```diff
@@ -1,10 +1,20 @@
...
+	e.POST("/test", controllers.ClusterTest)
+	e.POST("/testData", controllers.ClusterTestWithData)
+	e.POST("/populateMockData", controllers.PopulateClusterWithMockData)
```
[the remark this answers is not in the document]
Comment (NataliAharoniPayu) [reply server/router.go:19]: A 'setter' variable been added to the operator config, named 'ExposeSensitiveEntryPoints', it is set defaultly to false, and indicates the operator rather or not to allow the continuity of sensitive flows origined from all the sensitive entry points: 
* Cluster reset
* Cluster flush all data
* Cluster populate with mock data
* Cluster test with mock data

File: helm/values.yaml (hunk 4 above)
Comment (NataliAharoniPayu) [reply helm/values.yaml]: will do

**Pull request closed without merging**